### PR TITLE
Adding country support for Germany, retry in case of connection error, option for Janssen

### PR DESCRIPTION
**Pull request by edwillys**
Please feel free to comment on the changes and let me know if you would be willing to merge all or a part of the changes.

### CODE REVIEW

**Comment by rbignon** [comment]
Thank you for this nice contribution. Why changing 'username', 'password' and 'city' to required options?

I would prefer keeping positional arguments, and add --country as an optional parameter (with the default value guessed from the system locales).

**Comment by edwillys** [comment]
No worries. `password` is not a mandatory argument. I changed from positional to tagged to keep the consistency with the other options, as I found a bit strange to mix both styles. If you would like to keep it it mixed, I can revert that part.

As for --country, Are not sure about setting the default value to the system locale? Wouldn't this cause confusion in case the locale doesn't match the country the user wants to make an appointment? Also a possible tricky use case would be if a city name is available in multiple countries (can't think of one right now though)

File: README.md [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
 
-This script lets you automatically book a vaccine slot on Doctolib in France in
-the next seven days.
+This script lets you automatically book a vaccine slot on Doctolib in France and in Germany in
+the next seven days. 
 
 
-<p align="center">
-  <img src="https://raw.githubusercontent.com/rbignon/doctoshotgun/da5f65a1e2ecc7b543376b1549c62004a454b90d/example.svg">
-</p>
```
Comment (rbignon) [inline README.md:9]: You do not like this beautiful video? :)

**Comment by rbignon** [comment]
> No worries. `password` is not a mandatory argument. I changed from positional to tagged to keep the consistency with the other options, as I found a bit strange to mix both styles. If you would like to keep it it mixed, I can revert that part.

`city` and `email` are mandatory parameters, so it is better to be positional arguments (more user friendly imo).

> As for --country, Are not sure about setting the default value to the system locale? Wouldn't this cause confusion in case the locale doesn't match the country the user wants to make an appointment? Also a possible tricky use case would be if a city name is available in multiple countries (can't think of one right now though)

If you think it is better to be explicit, I prefer to make 'country' the first positional argument.

**Comment by edwillys** [comment]
Done. Only thing is, I didn't update the example GIF from the readme. As it wasn't up to date anymore I deleted it for now. I guess you might want to update it with a snippet from your computer (?)

**Comment by rbignon** [comment]
Hello,

I've rebased your branch on master and pushed on https://github.com/rbignon/doctoshotgun/tree/de, can you check it's ok for you?

**Comment by edwillys** [comment]
Hello, thanks for that. Yeah, it looks fine. Just one comment:
why not keep the ``` self.open( self.BASEURL + '/sessions/new') ``` 
```
    def do_login(self):
        self.open('https://www.doctolib.fr/sessions/new')
```

In the end the login works regardless of the level domain though.

**Comment by rbignon** [comment]
> Hello, thanks for that. Yeah, it looks fine. Just one comment:
> why not keep the `self.open( self.BASEURL + '/sessions/new')`
> 
> ```
>     def do_login(self):
>         self.open('https://www.doctolib.fr/sessions/new')
> ```
> 
> In the end the login works regardless of the level domain though.

It's an error during the conflict resolution, I fix it.

**Comment by rbignon** [comment]
Merged in master, thank you!

**Pull request closed without merging**